apoc.monitor.store fails for block stores

The following procedure call : `CALL apoc.monitor.store()` fails when database stores are in block format.

## Expected Behavior (Mandatory)

The procedure should return store statistics

## Actual Behavior (Mandatory)

```
Failed to invoke procedure `apoc.monitor.store`: Caused by: java.lang.IllegalArgumentException: PlainDatabaseLayout{databaseDirectory=/Users/cwillemsen/dev/_graphs/actro/data/databases/neo4j, transactionLogsDirectory=/Users/cwillemsen/dev/_graphs/actro/data/transactions/neo4j} does not describe a record storage database.
```

## How to Reproduce the Problem

1. Run Neo4j 5.23
2. Execute the following query in the browser : `CALL apoc.monitor.store()` 

## Versions

  - OS: N/A
  - Neo4j: 5.23
  - Neo4j-Apoc: 5.23

--- a/extended-it/src/test/java/apoc/monitor/StoreInfoProcedureEnterpriseTest.java
+++ b/extended-it/src/test/java/apoc/monitor/StoreInfoProcedureEnterpriseTest.java
@@ -1,0 +1,47 @@
+package apoc.monitor;
+
+import apoc.util.Neo4jContainerExtension;
+import apoc.util.TestContainerUtil;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.neo4j.driver.Session;
+
+import java.util.List;
+
+import static apoc.util.TestContainerUtil.createEnterpriseDB;
+import static apoc.util.TestContainerUtil.testCall;
+import static org.junit.Assert.assertNotNull;
+
+public class StoreInfoProcedureEnterpriseTest {
+    private static Neo4jContainerExtension neo4jContainer;
+    private static Session session;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        neo4jContainer = createEnterpriseDB(List.of(TestContainerUtil.ApocPackage.EXTENDED), true);
+        neo4jContainer.start();
+
+        session = neo4jContainer.getSession();
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        session.close();
+        neo4jContainer.close();
+    }
+
+    @Test
+    public void testGetStoreInfo() {
+        testCall(session, "CALL apoc.monitor.store()", (row) -> {
+            assertNotNull(row.get("logSize"));
+            assertNotNull(row.get("stringStoreSize"));
+            assertNotNull(row.get("arrayStoreSize"));
+            assertNotNull(row.get("nodeStoreSize"));
+            assertNotNull(row.get("relStoreSize"));
+            assertNotNull(row.get("propStoreSize"));
+            assertNotNull(row.get("totalStoreSize"));
+        });
+    }
+}

--- a/extended/src/main/java/apoc/monitor/Store.java
+++ b/extended/src/main/java/apoc/monitor/Store.java
@@ -27,7 +27,7 @@ public class Store {
 
         Database database = ((GraphDatabaseAPI) db).getDependencyResolver().resolveDependency(Database.class);
         //This will work only on Record format databases. Has to be updated when any additional format is available
-        RecordDatabaseLayout databaseLayout = RecordDatabaseLayout.cast( database.getDatabaseLayout() );
+        RecordDatabaseLayout databaseLayout = RecordDatabaseLayout.convert( database.getDatabaseLayout() );
         return Stream.of(new StoreInfoResult(
                 getDirectorySize(databaseLayout.getTransactionLogsDirectory().toFile()),
                 databaseLayout.propertyStringStore().toFile().length(),

